lib/vanagon/engine/scheduled.rb

```diff
@@ -0,0 +1,27 @@
+require 'vanagon/engine/base'
+require 'json'
+
+class Vanagon
+  class Engine
+    # Class to use when building on a shared resource device that is managed
+    # by an external scheduler.
+    class Scheduled < Base
```

Comment:
Going to dig into this a bit. There's a checklist at the top of the PR, let's not merge until that's complete.